botocore/args.py

```diff
@@ -464,3 +466,63 @@ def _ensure_boolean(self, val):
             return val
         else:
             return val.lower() == 'true'
+
+    def _compute_endpoint_resolver_v2_builtin_defaults(
```

Comment:
This is something I was going to adjust in David's PR, but I'm thinking we probably want to remove the "v2" designation from external functions. That will allow the interfaces to stay in place with future evolutions rather than requiring a v3 if there's ever a next iteration.

Maybe something like `_compute_endpoint_resolver_builtins`? Are they necessarily defaults, or the actual resolved values? I'm not sure if these are used elsewhere in future PRs.

Reply:
> Are they necessarily defaults, or the actual resolved values?

I added the `_defaults` suffix at the last minute here. In most cases, these are the actual values. The exception is the `AWS::S3::ForcePathStyle` builtin. To maintain full backwards compatibility, this will sometimes need to be changed from the default value based on operation-specific operation. 

Here is a specific example:

```python
import botocore.session
from botocore.config import Config

session = botocore.session.get_session()
client = session.create_client(
    's3',
    config=Config(s3={addressing_style='path'}),
)

client.get_object_attributes(
    Bucket="arn:aws:s3:us-west-2:1234567890:accesspoint/my-bucket-access-point",
    Key="object.txt",
    ObjectAttributes=["ETag", "Checksum"],
)
```

If we set `AWS::S3::ForcePathStyle = True` purely based on the value of `addressing_style` in the config, then the ruleset will return this error:

```
Path-style addressing cannot be used with ARN buckets
```

Botocore's current behavior is to silently ignore the `addressing_style` config when an ARN is used. To maintain this behavior, we need to change the `AWS::S3::ForcePathStyle` to `False` when the bucket name is an ARN. The bucket name can be different for each operation. This is why I decided to call the builtin values assigned at client construction time "defaults". `AWS::S3::ForcePathStyle` is the only builtin where operation-time modifications are necessary, at least for now.